src/lib/slide-toggle/slide-toggle-config.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {InjectionToken} from '@angular/core';
+
+
+/**
+ * Slide toggle click action when user click on input element.
+ * noop: Do not toggle checked.
+ * check: Only toggle checked status. Default behavior
+ * undefined: Same as `check`.
+ */
+export type MatSlideToggleClickAction = 'noop' | 'check' | undefined;
+
+/**
+ * Injection token that can be used to specify the slide toggle click behavior.
+ */
+export const MAT_SLIDE_TOGGLE_CLICK_ACTION =
+  new InjectionToken<MatSlideToggleClickAction>('mat-slide-toggle-click-action');
+
+/**
+ * Slide toggle drag action when user drag on slide toggle.
+ * noop: Do not toggle checked.
+ * check: Only toggle checked status. Default behavior
+ * undefined: Same as `check`.
+ */
+export type MatSlideToggleDragAction = 'noop' | 'check' | undefined;
```

Comment:
If someone sets this action to `noop`, wouldn't it make sense to just block dragging at all?

Also, does it make sense to call the other value `check`? Wouldn't `toggle` fit better here?

Reply:
The use case is when user toggle/drag the slide toggle, a dialog popup and requires user's confirmation. After the user confirmed the value is changed. 

I think we still need to enable dragging.

Changed to toggle